dipy/workflows/align.py

```diff
@@ -698,3 +699,188 @@ def run(self, static_image_file, moving_image_files, affine_matrix_file,
+             displacement field at each iteration.
+
+        ss_sigma_factor : float, optional
+            parameter of the scale-space smoothing kernel. For example, the
+             std. dev. of the kernel will be factor*(2^i) in the isotropic case
+             where i = 0, 1, ..., n_scales is the scale.
+
+        opt_tol : float, optional
+            the optimization will stop when the estimated derivative of the
+             energy profile w.r.t. time falls below this threshold.
+
+        inv_iter : int, optional
+            the number of iterations to be performed by the displacement field
+             inversion algorithm.
+
+        inv_tol : float, optional
+            the displacement field inversion algorithm will stop iterating
+             when the inversion error falls below this threshold.
+
+        out_dir : string, optional
+            Directory to save the transformed files (default '').
+
+        out_warped : string, optional
+            Name of the warped file. If no name is given then a
+             suffix 'transformed' will be appended to the name of the
+             original input file (default 'warped_moved.nii.gz').
+
+        out_inv_static : string, optional
+            Name of the file to save the static image after applying the
+             inverse mapping (default 'inv_static.nii.gz').
+
+        out_field : string, optional
+            Name of the file to save the diffeomorphic field.
+
+        """
+        io_it = self.get_io_iterator()
+        metric = metric.lower()
+        if metric not in ['ssd', 'cc', 'em']:
+            raise ValueError("Invalid similarity metric: Please"
+                             " provide a valid metric.")
+
+        # Init parameter if they are not setup
+        init_param = {'ssd': {'mopt_smooth': 4.0,
+                              'mopt_inner_iter': 10,
+                              'mopt_step_type': 'demons'
+                              },
+                      'em': {'mopt_smooth': 1.0,
+                             'mopt_inner_iter': 5,
+                             'mopt_step_type': 'gauss_newton'
+                             }
+                      }
+        mopt_smooth = mopt_smooth or init_param[metric]['mopt_smooth']
+        mopt_inner_iter = mopt_inner_iter or  \
+            init_param[metric]['mopt_inner_iter']
+        mopt_step_type = mopt_step_type or \
+            init_param[metric]['mopt_step_type']
+
+        for static_file, moving_file, in_affine, \
+                warped_file, inv_static_file, displ_file in io_it:
+
+            # Loading the image data from the input files into object.
+            static_image, static_grid2world = load_nifti(static_file)
+            moving_image, moving_grid2world = load_nifti(moving_file)
+
+            # Sanity check for the input image dimensions.
+            ImageRegistrationFlow.check_dimensions(static_image, moving_image)
```

Comment:
Maybe make the `check_dimensions` a module-wide utility function, rather than a method of the `ImageRegistrationFlow`?